src/lib/posts.js

```diff
@@ -19,15 +25,89 @@ export function postPathBySlug(slug) {
 
 export async function getPostBySlug(slug) {
   const apolloClient = getApolloClient();
+  const apiHost = new URL(process.env.WORDPRESS_HOST || process.env.WORDPRESS_GRAPHQL_ENDPOINT).host;
 
-  const data = await apolloClient.query({
-    query: getQueryPostBySlug(slug),
-  });
+  let postData;
+  let seoData;
+
+  try {
+    postData = await apolloClient.query({
+      query: QUERY_POST_BY_SLUG,
+      variables: {
+        slug,
+      },
+    });
+  } catch (e) {
+    console.log(`Failed to query post data: ${e.message}`);
+    throw e;
+  }
+
+  const post = [postData?.data.postBy].map(mapPostData)[0];
+
+  // If the SEO plugin is enabled, look up the data
+  // and apply it to the default settings
+
+  if (process.env.WORDPRESS_PLUGIN_SEO === true) {
```

Comment:
In this block, I was wondering if throwing error on the catch block is a "desirable" behaviour. Some thoughts on it:

When querying the seo of the post, if the execution breaks, below the custom error message it's shown a large Apollo error (if the plugin is marked as enabled and the WP instance is not prepared with the requirements). So it may be more difficult to locate the reason and solution as it is pushed up on the terminal by the "throwed" error. I was thinking If instead of throwing an error, the flow could continue (showing the custom error). Also the page which takes the data wont be broken if the user forgets to mark WORDPRESS_PLUGIN_SEO as false.

In case you think is a good idea, I found this [label statement](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/label) that could work for this case:

```
seoPlugin: 
  if (process.env.WORDPRESS_PLUGIN_SEO === true) {
    try {
      ....
    } catch (e) {
      ...
      break seoPlugin; //Instead of throw e
    }
...
```
This way the block is interrupted if there is an error, but the data without the seo is returned.